src/DDTrace/Integrations/Lumen/V5/LumenIntegrationLoader.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace DDTrace\Integrations\Lumen\V5;
+
+use DDTrace\Configuration;
+use DDTrace\GlobalTracer;
+use DDTrace\Integrations\Integration;
+use DDTrace\Integrations\Lumen\LumenIntegration;
+use DDTrace\Tag;
+use DDTrace\Type;
+use Symfony\Component\HttpFoundation\Request;
+
+final class LumenIntegrationLoader
+{
+    public function load()
+    {
+        if (!$this->shouldLoad()) {
+            return Integration::NOT_AVAILABLE;
+        }
+
+        $span = GlobalTracer::get()->getRootScope()->getSpan();
+        $span->overwriteOperationName('lumen.request');
+        $span->setTag(Tag::SERVICE_NAME, $this->getAppName());
+        $span->setIntegration(LumenIntegration::getInstance());
+        $span->setTraceAnalyticsCandidate();
+
+        // prepareRequest() was added in Lumen 5.2
+        // https://github.com/laravel/lumen-framework/blob/5.2/src/Application.php#L440
+        dd_trace('Laravel\Lumen\Application', 'prepareRequest', function (Request $request) use ($span) {
+            $span->setTag(Tag::HTTP_URL, $request->getUri());
+            $span->setTag(Tag::HTTP_METHOD, $request->getMethod());
+            return dd_trace_forward_call();
+        });
+
+        dd_trace('Laravel\Lumen\Application', 'dispatch', function () use ($span) {
+            $response = dd_trace_forward_call();
+            $resourceName = 'unnamed_route';
+            if (isset($this->currentRoute[1]['uses'])) {
```

Comment:
Good question - I can confirm that this [appears to be the case since 5.2](https://github.com/laravel/lumen-framework/blob/5.2/src/Concerns/RoutesRequests.php#L393) so we should be good! :)